src/utils/parse-arguments-into-options.ts

```diff
@@ -1,11 +1,30 @@
-import type { Args, RawOptions } from "../types";
+import type { Args, ExternalExtension, RawOptions } from "../types";
 import arg from "arg";
 import * as https from "https";
 import { getDataFromExternalExtensionArgument } from "./external-extensions";
 import chalk from "chalk";
 import { CURATED_EXTENSIONS } from "../config";
+import fs from "fs";
+import path from "path";
+import { fileURLToPath } from "url";
+
+const validateTemplate = async (
+  template: string,
+  dev: boolean,
+): Promise<{ repository: string; branch?: string } | string> => {
+  if (dev) {
+    // Check that the template folders exists in extensions/${template}
+    try {
+      const currentFileUrl = import.meta.url;
+      const extensionsDirectory = path.resolve(decodeURI(fileURLToPath(currentFileUrl)), "../../externalExtensions");
+      await fs.promises.access(`${extensionsDirectory}/${template}`);
+    } catch {
+      throw new Error(`Template not found in "externalExtensions/${template}"`);
+    }
+
+    return template;
+  }
```

Comment:
Umm I think we should call this function `validateExternalExtension` ? 

also the name of first arg from `template` => `extensionName`

just wondering why we used template naming convention lol sorry if I am missing something too obvious

Reply:
Yes! I think that **template** was the original name for the extensions, but then we changed it.

Reply:
Ohhh yess, thanks! just remembered it lol it was a main discussion in #32

Reply:
Thank you!! https://github.com/scaffold-eth/create-eth/pull/57/commits/81c6990c167521e5e801e8e4abe8f9edab73d757